app/scripts/lib/channels/inter-tab.js

```diff
@@ -93,19 +104,44 @@ define([
     },
 
     on: function (name, callback) {
-      // crosstab registers callbacks by storing functions by a list
-      // of keys. If no key is given, a key is generated by converting the
-      // function to a string. This means a given function can only be
-      // registered once unless a key is also passed in during registration.
-      // Use the current date plus a random number in case two functions are
-      // reigstered in the same millisecond.
-      var key = Date.now() + Math.random();
-      this._crosstab.util.events.on(name, callback, key);
-      return key;
+      this._handlers[name] = this._handlers[name] || [];
+      var sentMessageIds = this._sentMessageIds;
+
+      // crosstab sends notifications to the the current tab as well
+      // as any remote tabs. This behavior is not what we want. Wrap
+      // the supplied callback that checks whether the current tab sent
+      // the event, if so, do not call the callback.
+      function wrapperCallback(event) {
```

Comment:
Purely cosmetic but I'm not a big fan of placing inner functions declarations part-way down their parent. I prefer to lean on hoisting and place them at the bottom, as it makes for a more top-down code order. That may be a weird preference in which case ignore it, but just thought I'd mention it anyway.